lib/engine/game/base.rb

```diff
@@ -1983,35 +1985,60 @@ def ability_right_owner?(entity, ability, owner_type)
           when nil
             true
           end
-        return false unless correct_owner_type
-        return false if owner_type && (ability.owner_type.to_s != owner_type.to_s)
 
-        true
+        !!correct_owner_type
       end
 
       def ability_usable_this_or?(ability)
         !ability.count_per_or || (ability.count_this_or < ability.count_per_or)
       end
 
-      def ability_right_time?(ability, time, strict_time)
-        return false if strict_time && !ability.when
-        return true unless time
+      def ability_right_time?(ability, time, on_phase, passive_ok)
```

Comment:
The changes in this function are the main point of this PR

Reply:
should time any be default?

Reply:
I think no, using any as the default opens up more possibilities for abilities to be used out-of-turn before the check for `other_or` is done